### PR TITLE
SRP API changes

### PR DESCRIPTION
**Pull request by pbbastian**
**DO NOT MERGE YET** (but please do review)
Needs PR 64101 to go into `trunk`

This PR updates the code to work with the forthcoming changes to the SRP API based on an API review we got from the scripting team.

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
Adding @uygar3d for awareness of shadow change

**Comment by sebastienlagarde** [comment]
@peterbay is this code in 2018.2? Or in trunk? If 2018.2 are we ready to merge now?

**Comment by sebastienlagarde** [comment]
(I think we need to be sure that not inbetween PR have change code also)

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs [hunk 1]
```diff
@@ -433,8 +433,7 @@ void InitializeRenderStateBlocks()
         {
             m_DepthStateOpaque = new RenderStateBlock
             {
-                depthState = new DepthState(true, CompareFunction.LessEqual),
-                mask = RenderStateMask.Depth
```
Comment (JulienIgnace-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs]: Why is the mask gone here?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs [hunk 2]
```diff
@@ -704,10 +703,10 @@ public override void Render(ScriptableRenderContext renderContext, Camera[] came
                                         if (m_Asset.renderPipelineSettings.lightLoopSettings.skyLightingOverrideLayerMask == -1)
                                             layerMask = -1;
                                         else
-                                            layerMask = (-1 & ~m_Asset.renderPipelineSettings.lightLoopSettings.skyLightingOverrideLayerMask);
-                                    }
+                                    layerMask = (-1 & ~m_Asset.renderPipelineSettings.lightLoopSettings.skyLightingOverrideLayerMask);
```
Comment (JulienIgnace-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs]: Seems like indentation broke here.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs [hunk 3]
```diff
@@ -755,9 +754,10 @@ public override void Render(ScriptableRenderContext renderContext, Camera[] came
 
                     ReflectionSystem.PrepareCull(camera, m_ReflectionProbeCullResults);
 
+                    CullResults cullResults;
```
Comment (JulienIgnace-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs]: We want to keep this as a member to be able to implement the "Freeze Visibility" debug mode (where we need to stop updating the culling to "freeze" the camera culling in a given state)
It's not in yet, but Torbjorn tested it a while ago and we should be able to bring it back to master soon enough.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs [hunk 4]
```diff
@@ -1654,10 +1652,10 @@ public void PushColorPickerDebugTexture(CommandBuffer cmd, RenderTargetIdentifie
             {
                 using (new ProfilingSample(cmd, "Push To Color Picker"))
                 {
-                    HDUtils.BlitCameraTexture(cmd, hdCamera, textureID, m_DebugColorPickerBuffer);
-                }
+                HDUtils.BlitCameraTexture(cmd, hdCamera, textureID, m_DebugColorPickerBuffer);
```
Comment (JulienIgnace-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs]: Indentation

**Pull request closed without merging**